autotest/OldTests/src/test/java/retry/FailureRetryAnalyzer.java

```diff
@@ -0,0 +1,38 @@
+package retry;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.testng.IRetryAnalyzer;
+import org.testng.ITestResult;
+
+public class FailureRetryAnalyzer implements IRetryAnalyzer {
+  int currentRetry = 0;
+  Logger LOGGER = Logger.getLogger(FailureRetryAnalyzer.class.getName());
+
+  @Override
+  public boolean retry(ITestResult result) {
+    MaxRetryCount failureRetryCount =
+        result.getMethod().getConstructorOrMethod().getMethod().getAnnotation(MaxRetryCount.class);
+    int maxRetryCount = (failureRetryCount == null) ? 0 : failureRetryCount.value();
+    if (++currentRetry > maxRetryCount) {
+      currentRetry = 0;
+      return false;
+    } else {
+      logRetryInfo(result, maxRetryCount);
+      return true;
+    }
+  }
+
+  private void logRetryInfo(ITestResult result, int maxRetryCount) {
+    LOGGER.warning("Test with retry analyser failed.\n\n");
+    // print failure stack trace
+    LOGGER.log(Level.WARNING, result.getThrowable().getMessage(), result.getThrowable());
```

Comment:
```suggestion
    LOGGER.warning(result.getThrowable().getMessage(), result.getThrowable());
```

or can you

```suggestion
    LOGGER.warning(result.getThrowable());
```

Actually, do we even need to log this? Seems excessively verbose.

Reply:
Your suggestion will not work. 
LOGGER.warning() helper function doesn't have a stack trace method like LOGGER.error does. In order to get a stack trace, and a warning label, you must LOGGER.log(level, message, throwable)

Reply:
The purpose here was so that when a TestNG method fails and retries, the initial failure will still log a stack trace.

Reply:
> Your suggestion will not work.

Yup, as it appears we're not using log4j, just vanilla JRE logging. We should be using log4j

> The purpose here was so that when a TestNG method fails and retries, the initial failure will still log a stack trace.

But what does that offer when we already have the report that test t from class c is retrying? We don't want our logs for testing even more verbose unless it serves a purpose.

Reply:
The idea I had for that is that if some test is flakey, we can look into why it was breaking without letting it fail the whole run. But if you don't think that's worthwhile I'm happy to remove it.

Reply:
Okay, so as discussed just now, this is to capture initial details of a test that eventually passes but we may want to go back at a later date an look into the flakiness.

Seeing this would really be the exception to the norm, let's log the stacktrace at 'debug' once the logger is changed over to log4j.